server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1427,9 +1405,13 @@ public void run() {
 
     final long deadline = System.currentTimeMillis() + MAX_CLEANUP_WAIT_TIME;
     statusThread.join(remaining(deadline));
-    replicationWorkAssigner.join(remaining(deadline));
-    replicationWorkDriver.join(remaining(deadline));
-    replAddress.server.stop();
+    if (!getConfiguration().get(Property.REPLICATION_NAME).isEmpty()) {
```

Comment:
What about always checking if these are non-null and stopping them if they are?

Checking the configuration for the property (while it should always be accurate) has the potential to introduce a bug where we could just check everything and be certain it's all stopped.

Reply:
Ahhh I see.  Yes, that would be better.  Good idea.